### ISSUE TITLE
core: provide module that acts as a drop-in replacement for MLIR Python bindings

### ISSUE THREAD

**Issue by superlopuh** [xdslproject/xdsl#913]
This would let us test out functionality and compatibility with MLIR. It seems like one of the easiest ways to write tests for the things we'd like to mirror exactly.

**Comment by webmiche** [comment]
I don't quite get what you mean. You mean we should provide something xdsl-backed that people could install and the use their previous MLIR python bindings code with? So a package that mirrors the mlir.ir import structure of the python bindings?

**Comment by webmiche** [comment]
I think this is a cool idea especially in a transitioning period from python bindings to xDSL.
This would also help us understand the bindings more shortcomings/advantages that they have. I feel one of the dangers of this is that we start designing parts of xdsl with this module in mind limiting us in what actually makes sense for xdsl.

**Comment by superlopuh** [comment]
yep, exactly, something where replacing `import mlir` with `import xdsl_bindings as mlir` lets the rest of the code run unmodified with the same behaviour

**Comment by superlopuh** [comment]
> This would also help us understand the bindings more shortcomings/advantages that they have. I feel one of the dangers of this is that we start designing parts of xdsl with this module in mind limiting us in what actually makes sense for xdsl.

Depends on how many shortcuts we take implementing it :)
